src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -5883,26 +5883,82 @@ public void ToString_DerivedRecord_AbstractSealed()
             Assert.True(toString.IsImplicitlyDeclared);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void ToString_DerivedRecord_BaseHasSealedToString(bool usePreview)
+        {
+            var src = @"
+var c = new C2();
+System.Console.Write(c.ToString());
+
+record C1
+{
+    public sealed override string ToString() => ""C1"";
+}
+record C2 : C1;
+";
+
+            var comp = CreateCompilation(src, parseOptions: usePreview ? TestOptions.RegularPreview : TestOptions.Regular9, options: TestOptions.DebugExe);
+            if (usePreview)
+            {
+                comp.VerifyEmitDiagnostics();
+                CompileAndVerify(comp, expectedOutput: "C1");
+            }
+            else
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (4,35): error CS8652: The feature 'sealed ToString in record' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     public sealed override string ToString() => throw null;
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "ToString").WithArguments("sealed ToString in record").WithLocation(7, 35),
+                    // (9,8): error CS8912: Inheriting from a record with a sealed 'Object.ToString' is not supported in C# 9. Please use language version 'preview' or greater.
```

Comment:
>// (9,8): error CS8912: Inheriting from a record with a sealed 'Object.ToString' is not supported in C# 9. Please use language version 'preview' or greater. [](start = 20, length = 156)

In this scenario the error feels somewhat redundant. In cases like this, we usually supress it. For example, if the base that seals Object.ToStrung is declared in the same module, we can skip reporting the error. VerifyOverridesMethodFromObject helper does something similar.

Reply:
Done.
I'd like to add a test to ensure that the warning is emitted if the base that seals `ToString` *isn't* in the same module, but I'm not sure how to do that... I could probably do it with IL source, but that seems overkill; maybe there's an easier way?

Reply:
>I could probably do it with IL source, but that seems overkill; maybe there's an easier way?

It is possible to reference one compilation from another. See Inheritance_03 unit-test for example.

---
In reply to: [604865992](https://github.com/dotnet/roslyn/pull/52029#discussion_r604865992) [](ancestors = 604865992)

Reply:
This scenario is covered in `ToString_BadBase_SealedToString`